### PR TITLE
Intercept memalign and pvalloc

### PR DESCRIPTION
**Pull request by davidtrevelyan**
Fixes https://github.com/realtime-sanitizer/radsan/issues/26

### CODE REVIEW

**Review by cjappl — APPROVED** [review]

File: compiler-rt/lib/radsan/tests/radsan_test_interceptors.cpp [hunk 1]
```diff
@@ -116,6 +120,22 @@ TEST(TestRadsanInterceptors, posixMemalignDiesWhenRealtime) {
...
+  expectRealtimeDeath(func, "memalign");
+  expectNonrealtimeSurvival(func);
+}
+#endif
+
+#if SANITIZER_INTERCEPT_PVALLOC
+TEST(TestRadsanInterceptors, pvallocDiesWhenRealtime) {
+  auto func = []() { EXPECT_NE(pvalloc(2048), nullptr); };
+  expectRealtimeDeath(func, "pvalloc");
+  expectNonrealtimeSurvival(func);
```
Comment (cjappl) [inline compiler-rt/lib/radsan/tests/radsan_test_interceptors.cpp:135]: Do we have a general sense of what we want the "full compilation test" vs these unit tests? When is it necessary to have a full lit test vs just having these unit tests?

May be good to standardize around when we think both are necessary. (this is outside of the scope of this review, and does not block it)

Comment (davidtrevelyan) [reply compiler-rt/lib/radsan/tests/radsan_test_interceptors.cpp:135]: Good question - I think this warrants a proper discussion. Maximising both test coverage and maintainability is a delicate art form! I do have an opinion that is best expressed in the form of some analogies - but it'll be easiest to explain on a call. Let's schedule a time!

Comment (cjappl) [reply compiler-rt/lib/radsan/tests/radsan_test_interceptors.cpp:135]: I'll add it to the list for our next chat :)

Comment (davidtrevelyan) [reply compiler-rt/lib/radsan/tests/radsan_test_interceptors.cpp:135]: Awesome. Looking forward to hearing your opinions too of course!
